### DISCUSSION TITLE
Homematic Geräte tauchen nicht bei den integrationen auf

### DISCUSSION THREAD

**Discussion by ThorstenHalsch** [OpenCCU/OpenCCU#1549] [🇩🇪 Fragen&Antworten]
Hallo, vieleicht kann mir jemand weiter helfen.
Ich versuche meine Homematic Komponenten im Home Assistant zu laufen zu bekommen. Bei der Installation bin ich nach folgender Berschreibung vorgegangen:
https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant
RasperryMatic läuft und ist auch mit meinen Devices verbunden.
Die geräte tauchen aber im Home Assistant nirgens auf. 
Ich vermute das es an den IP Addressen in der configuration.yaml liegt. In der ccu Oberfläche sehe ich keine IP. Ich kann nur einen Hostname setzen. Mir ist aber nicht klar ob der Docker Container über einen DHCP eine IP zugewiesen bekommt. Verwende ich diesen Hostnamen bekomme ich folgenden Fehler

Logger: homeassistant.setup
Source: components/homematic/__init__.py:253
First occurred: 21:17:25 (1 occurrences)
Last logged: 21:17:25
Error during setup of component homematic

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pyhomematic/connection.py", line 76, in start
    self._server.start()
AttributeError: 'HMConnection' object has no attribute '_server'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 254, in _async_setup_component
    result = await task
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/homematic/__init__.py", line 253, in setup
    homematic.start()
  File "/usr/local/lib/python3.9/site-packages/pyhomematic/connection.py", line 82, in start
    self._server.stop()
AttributeError: 'HMConnection' object has no attribute '_server'

So sieht meine Konfiguration aus

# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

homematic:
  interfaces:
    rf:
      host: RaspiMatic
      port: 2001
      resolvenames: json
      username: Admin
      password: xxx
    wired:
      host: RaspiMatic
      port: 2000
      resolvenames: json
      username: Admin
      password: xxx
    hmip:
      host: RaspiMatic
      port: 2010
      resolvenames: json
      username: Admin
      password: xxx
    groups:
      host: RaspiMatic
      port: 9292
      resolvenames: json
      username: Admin
      password: xx
      path: /groups
  hosts:
    raspberrymatic:
      host: RaspiMatic
      username: Admin
      password: xxx
      
sensor:
  - platform: template
    sensors:
      raspberrymatic_last_reboot:
        value_template: "{{ state_attr('homematic.raspberrymatic', 'V_Last_Reboot') or '01.01.1970 00:00:00' }}"
        icon_template: "mdi:clock"
        entity_id: homematic.raspberrymatic

![Network](https://user-images.githubusercontent.com/72215618/142687864-fb708f6f-b32c-45a3-9729-3db1197dccfc.JPG)

**Comment by Baxxy13** [comment]
In der `configuration.yaml` trägt man doch den Hostnamen ein der innerhalb HA in den RM-AddOn Informationen steht.
Hier mal der Hostname eines RM-Nightlys.
![RM-HA-AddOn_Hostname](https://user-images.githubusercontent.com/54898249/142697064-c535e7e5-9f8a-420c-bbcc-2dee991f3271.JPG)
Also genau so wie es im [Wiki](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#add-the-raspberrymatic-addon-to-configurationyaml) steht.

**Reply by ThorstenHalsch** [reply]
Hi, danke für die schnelle Antwort
Leider funktioniert das auch nicht. Das war mei erster Versuch:

![grafik](https://user-images.githubusercontent.com/72215618/142725849-247aac1f-9e35-47d5-ba5b-04ac64c255f9.png)
![grafik](https://user-images.githubusercontent.com/72215618/142725804-3a02777a-d6b3-48b5-ae46-b5b29a1062ef.png)

Ich dachte das danach RasperryMatic hier auftaucht.
![grafik](https://user-images.githubusercontent.com/72215618/142725877-f5885393-bffc-43aa-ad12-54f8f17e4480.png)

**Comment by Baxxy13** [comment]
> Ich dachte das danach RasperryMatic hier auftaucht.

Ne, da kommt nix. Wenn es korrekt implementiert ist sollten die Geräte in HA unter Einstellungen > Entitäten auftauchen.

Du kannst auch mal versuchen alle Strings (Hostname, User, Passwort usw.) in " " zu setzen. Siehe: [EXAMPLE CONFIGURATION WITH MULTIPLE PROTOCOLS AND SOME OTHER OPTIONS SET](https://www.home-assistant.io/integrations/homematic/#example-configuration-with-multiple-protocols-and-some-other-options-set)

**Reply by ThorstenHalsch** [reply]
Hi, das hat leider auch nicht geholfen. In den Entitäten tauchen nur die Standard RaspberryMatic Geräte auf.
![grafik](https://user-images.githubusercontent.com/72215618/142736945-80abc524-25d3-4fee-9281-ce7aae6d182d.png)
Muss in der ccu ggf noch was eingestellt werden. Noch eine Idee?

**Comment by ThorstenHalsch** [comment]
Hi, nachdem ich die Sicherheitseinstellungen auf Relaxed gesetzt habe findet er meine Geräte
![grafik](https://user-images.githubusercontent.com/72215618/142737501-4bb8fdfd-caba-4285-9ac4-b7b8eacc550d.png)

Hab aber keine Ahnung was für die Sicherheit des kompletten Systems bedeutet

**Comment by Baxxy13** [comment]
Ah ja, die Firewall. Wird auch auf der oben verlinkten Seite erwähnt.
>Since CCU Version 3, the internal firewalls are enabled by default. You have to grant full access for the XML-RPC API or specify the IP-address of the Home Assistant instance and allowlist it, inside the CCU’s security settings.
